packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -364,168 +389,171 @@ contract OPContractsManagerGameTypeAdder is OPContractsManagerBase {
+            for (uint256 j = 0; j < gameTypes.length; j++) {
+                GameType gameType = gameTypes[j];
+
+                // Get the existing game implementation.
+                IFaultDisputeGame existingGame = IFaultDisputeGame(address(getGameImplementation(dgf, gameType)));
+
+                // If no implementation exists, skip.
+                if (address(existingGame) == address(0)) {
+                    continue;
+                }
+
+                // Grab the existing game constructor params and init bond.
+                IFaultDisputeGame.GameConstructorParams memory gameParams = getGameConstructorParams(existingGame);
+
+                // Create a new game input with the updated prestate.
+                OPContractsManager.AddGameInput memory input = OPContractsManager.AddGameInput({
                     disputeAbsolutePrestate: _prestateUpdateInputs[i].absolutePrestate,
-                    saltMixer: saltMixer,
+                    saltMixer: reusableSaltMixer(_prestateUpdateInputs[i]),
                     systemConfig: _prestateUpdateInputs[i].systemConfigProxy,
                     proxyAdmin: _prestateUpdateInputs[i].proxyAdmin,
-                    delayedWETH: IDelayedWETH(payable(address(fdgParams.weth))),
-                    disputeGameType: fdgParams.gameType,
-                    disputeMaxGameDepth: fdgParams.maxGameDepth,
-                    disputeSplitDepth: fdgParams.splitDepth,
-                    disputeClockExtension: fdgParams.clockExtension,
-                    disputeMaxClockDuration: fdgParams.maxClockDuration,
-                    initialBond: initBond,
-                    vm: fdgParams.vm,
-                    permissioned: false
+                    delayedWETH: IDelayedWETH(payable(address(gameParams.weth))),
+                    disputeGameType: gameParams.gameType,
+                    disputeMaxGameDepth: gameParams.maxGameDepth,
+                    disputeSplitDepth: gameParams.splitDepth,
+                    disputeClockExtension: gameParams.clockExtension,
+                    disputeMaxClockDuration: gameParams.maxClockDuration,
+                    initialBond: dgf.initBonds(gameType),
+                    vm: gameParams.vm,
+                    permissioned: gameType.raw() == GameTypes.PERMISSIONED_CANNON.raw()
+                        || gameType.raw() == GameTypes.SUPER_PERMISSIONED_CANNON.raw()
                 });
-            }
 
-            // Game inputs must be ordered with increasing game type values. So FDG is first if it exists.
-            if (hasFDG) {
-                inputs[0] = fdgInput;
-                inputs[1] = pdgInput;
-            } else {
-                inputs[0] = pdgInput;
+                // Add the new game type with updated prestate
+                OPContractsManager.AddGameInput[] memory inputs = new OPContractsManager.AddGameInput[](1);
+                inputs[0] = input;
+                addGameType(inputs);
             }
```

Comment:
OK, so `addGameType()` is called within the new loop. Took me a second, but this is a nice change.